ui/apps/platform/src/Containers/Collections/CollectionsPage.tsx

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import useURLParameter from 'hooks/useURLParameter';
 
 import CollectionsTablePage from './CollectionsTablePage';
+import CollectionsFormPage from './CollectionFormPage';
+import { parsePageActionProp } from './collections.utils';
 
 function CollectionsPage() {
     // TODO Implement permissions once https://issues.redhat.com/browse/ROX-12619 is merged
     // const { hasWriteAccess } = usePermissions();
     const hasWriteAccessForCollections = true; // hasWriteAccess('TODO');
 
+    const [pageAction, setPageAction] = useURLParameter('action', undefined);
+    const { collectionId } = useParams();
+    const validPageActionProp = parsePageActionProp(pageAction, collectionId);
+
+    useEffect(() => {
+        // If the URL structure somehow gets into a state with an invalid action, clear
+        // the parameter to avoid confusing the user.
```

Comment:
This reminds me to ask if search query has page action, is anything else valid? For example, search filter.

A counter-example, in case it might ever arise in the future, is generate policy from search.

At the moment risk transfers the policy via Redux store, but it will soon pass via search query, to minimize the coupling between the containers (plus opening the door to generate from other contexts).

```text
/main/risk?s[Add%20Capabilities]=NET_BIND_SERVICE
/main/policy-management/policies?action=generate&s[Add%20Capabilities]=NET_BIND_SERVICE

Reply:
I think an action with a search query makes sense as a valid use case (at least for policies, not collections so far). This was more addressing the (slim) possibility of someone visiting a page with a collection ID _plus_ action=create. These combinations aren't restricted on the policies page, and even though I think the only way for someone to get there is to hand-edit the URL, it does place the page into a funky state where the form doesn't quite work correctly.

Reply:
Yeah, I can believe that I overlooked an inconsistent state for policies.